src/MessageCreationUtil.js

```diff
@@ -1,15 +1,22 @@
 import crypto from 'crypto'
+import NodeCache from 'node-cache'
 import randomstring from 'randomstring'
 import { MessageLayer } from 'streamr-client-protocol'
 import { ethers } from 'ethers'
+import Stream from './rest/domain/Stream'
 
 const { StreamMessage } = MessageLayer
 
 export default class MessageCreationUtil {
-    constructor(auth, signer, userInfoPromise) {
+    constructor(auth, signer, userInfoPromise, getStreamFunction) {
```

Comment:
4 arguments is usually where I draw the line and  switch to named arguments i.e. options object, might be too much to change though.